sparse_autoencoder/metrics/histogram_metric.py

```diff
@@ -0,0 +1,41 @@
+"""L0 (sparsity) norm metric."""
+from collections import OrderedDict
+from typing import Any, final
+
+import wandb
+
+from sparse_autoencoder.metrics.train.abstract_train_metric import (
+    AbstractTrainMetric,
+    TrainMetricData,
+)
+
+
+@final
+class HistogramMetric(AbstractTrainMetric):
+    """Histogram metric — log the histogram, as well as the number of dead features."""
+
+    @final
+    def __init__(
+        self,
+        dead_threshold: float = 0.0,
+    ) -> None:
+        """Create a histogram metric.
+
+        Args:
+            dead_threshold: Threshold for considering a feature to be dead.
+            log_progress_bar: Whether to log the metric to the progress bar.
+            log_weights_and_biases: Whether to log the metric to Weights and Biases.
+        """
+        super().__init__()
+        self.dead_threshold = dead_threshold
+
+    @final
+    def calculate(self, data: TrainMetricData) -> OrderedDict[str, Any]:
+        """Create a log item for Weights and Biases."""
+        feature_mean_acts = data.learned_activations.mean(dim=0)
+        num_dead_features = (feature_mean_acts <= self.dead_threshold).sum().item()
```

Comment:
I think this is meaningless with such a small sample size (one batch). However if you just log a table of all neurons (and the indices of these) it may be possible to then do the analysis on wandb? Might be too much data to send.

Alternatively we may need to make this a "resample neurons" metric and log it there rather than for every train batch. Have you experimented with logging this on wandb to see what it looks like?